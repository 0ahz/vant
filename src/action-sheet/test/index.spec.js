import { mount } from '@vue/test-utils';
import { later } from '../../../test';
import ActionSheet from '..';

test('should render content after disabling the lazy-render prop', async () => {
  const wrapper = mount(ActionSheet);
  expect(wrapper.find('.van-action-sheet__content').exists()).toBeFalsy();

  wrapper.setProps({ lazyRender: false });
  await later();
  expect(wrapper.find('.van-action-sheet__content').exists()).toBeTruthy();
});

test('should render default slot and match snapshot', () => {
  const wrapper = mount(ActionSheet, {
    props: {
      show: true,
      title: 'Title',
    },
    slots: {
      default: () => 'Default',
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('should allow to use the teleport prop', () => {
  const root = document.createElement('div');
  mount(ActionSheet, {
    propsData: {
      show: true,
      teleport: root,
    },
  });

  expect(root.querySelector('.van-action-sheet')).toBeTruthy();
});

test('should have "van-popup--round" class when setting the round prop', () => {
  const wrapper = mount(ActionSheet, {
    propsData: {
      show: true,
      round: true,
    },
  });

  expect(wrapper.find('.van-popup--round').exists()).toBeTruthy();
});
