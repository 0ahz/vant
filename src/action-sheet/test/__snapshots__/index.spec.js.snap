// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default slot and match snapshot 1`] = `
<transition-stub>
  <div class="van-overlay">
    <!---->
  </div>
</transition-stub>
<transition-stub>
  <div class="van-popup van-popup--round van-popup--bottom van-action-sheet">
    <div class="van-action-sheet__header">Title<i class="van-badge__wrapper van-icon van-icon-cross van-action-sheet__close">
        <!---->
        <!---->
        <!----></i></div>
    <!---->
    <div class="van-action-sheet__content">
      <!---->Default</div>
    <!---->
    <!---->
  </div>
</transition-stub>
`;
